po/uz.po

```diff
@@ -134,23 +134,23 @@ msgstr "Implicit konvertatsiyalar"
 
 #: src/SUMMARY.md:38
 msgid "Arrays and for Loops"
-msgstr "Arraylar va for Loops"
+msgstr "Arraylar va for Looplar"
 
 #: src/SUMMARY.md:40
 msgid "Day 1: Afternoon"
-msgstr "Day 1: Afternoon"
+msgstr "1-kun: Tushdan keyin"
 
 #: src/SUMMARY.md:42
 msgid "Variables"
 msgstr "O'zgaruvchilar"
 
 #: src/SUMMARY.md:43
 msgid "Type Inference"
-msgstr "Type Inference"
+msgstr "Inference Turi"
```

Comment:
This should be "Tur Inferencei" or something similar that makes "Inference" or its translated substitute the head, right? Because it does not speak about a "Type of Inference" but vaguely "Inference of Types". I believe there are a few flips like this.

Reply:
Sadly, Uzbek language doesn't have any translations for complex & scientific words and therefore we decided to translate only suffixes of words and adopt the base. In this example below, it would be way more correctly to translate it as "Inferensiya Turi". I do apologize for untranslated parts, @ismoilovdevml has been bugging out with this and currently we are working on single scientific words dictionary.

Reply:
@katsuki-yuri The "-i" suffix in "Inferensiya Turi" is possessive person marker, right? Like "Inferensiyaning Turi". If that is the case, it does not produce the correct noun phrase, as in, English would be "The type of inference", which makes inference sound like a name of an object rather than a process.

"Tur Inferensiyasi" on the other hand produces the correct noun phrase.

Thank you very much for your kind reply

Reply:
(just reminder that original phrase is "Type Inference" meaning "Inference of Types" or "Types' Inference" roughly)

Reply:
@mehmetoguzderin thank you for your activity. But the problem is that we can't say  "Turlar xulosasi" elsewhere, because if we do, no one will understand. but if we say  "Inferensiya Turi"  everyone will understand. Translating everything in Uzbek will lead to mistakes. New terms adapted to information technology mainly in English appear mainly in English. When translated from English to other languages, it may not mean the same as in English. It's a mistake for us to translate everything raw, people don't understand it and it causes several problems and difficulties.

Reply:
I'd like to mention that many words in Uzbek language were adopted from other languages without an exact rule and the grammar is still incomplete even if it seems complete. When we tried to digitalize Uzbek language into reusable algorithm (which happened at https://github.com/uzinfocom-org/korrektor and many other projects), we faced many issues whether there were no proper rule or instruction to translate scientific or any kind other kind of words. Apart from that, there doesn't exist any "correct" rule for any part of Uzbek language as we have 12 variations of this language on every state of this country.

In a few words, our language is in **"the most fucked up"** state and we are trying to investigate more on this. The only thing we were left to do is create our own rules and grammar and force it to our country and currently this is what we are busy with.

Reply:
@katsuki-yuri I have no objection to the choice of vocabulary but rather to their construction in syntax and, through that, their preservation of semantics. Though if there's a history of this flipped phrase becoming some sort of norm, that's something else. But where possible, it could help people to have a better choice here. The diagram I attach to this review is the most I can do (although I think the Uzbek language is well-studied and has many properly functioning grammar descriptions, I can understand the lack of digital tooling to continuously validate that with kind words, and furthermore, lack of cross-linguistic equivalence efforts).
<img width="719" alt="5AF50040-39C6-4293-A523-60B6FA292DC0" src="https://user-images.githubusercontent.com/5333693/223801203-c12fcc15-83ef-43dd-bee1-131f82979037.png">

Reply:
> although I think the Uzbek language is well-studied and has many properly functioning grammar descriptions

You would say vise if you were a citizen of this country like me... All studies are from 2000's done by USSR linguists which can't be used with current Uzbek language and we don't have anything except third party vocabularies...

Reply:
> @katsuki-yuri I have no objection to the choice of vocabulary but rather to their construction in syntax and, through that, their preservation of semantics. Though if there's a history of this flipped phrase becoming some sort of norm, that's something else. But where possible, it could help people to have a better choice here. The diagram I attach to this review is the most I can do (although I think the Uzbek language is well-studied and has many properly functioning grammar descriptions, I can understand the lack of digital tooling to continuously validate that with kind words, and furthermore, lack of cross-linguistic equivalence efforts).
> <img width="719" alt="5AF50040-39C6-4293-A523-60B6FA292DC0" src="https://user-images.githubusercontent.com/5333693/223801203-c12fcc15-83ef-43dd-bee1-131f82979037.png">
> 

Do you know the Uzbek language?

Reply:
> ... and we don't have anything except third party unverified vocabularies...

For syntactic concerns, there's the great recently published & open-access reference of Turkic by Lars Johanson from Cambridge University Press. https://www.cambridge.org/core/books/turkic/D7BFED20E2E35B601B23B165967A8E28

@yetimdasturchi I'm familiar with decomposition of noun phrases at least. Otherwise, ad-hominem is not very helpful.

Reply:
(Chapters of the aforementioned textbook are no longer open-access, unfortunately, apologies about that)

Reply:
My father is Turkish.  I know Turkish and Uzbek well.  Better not to argue about it.  All your versions above are wrong.  Uzbek language rules are different from Turkish.  One of the main reasons for this is that the Uzbek language includes not only Turkish, but also Arabic, Persian and many European languages.  Examples from Turkic languages ​​are rarely used at the moment.

Reply:
I'm gonna repeat, all resource available on the internet about Uzbek language has no usage at all or can't be used with current Uzbek language which has no any proper rule or grammar instructions. Yes, Uzbek language is currently (the one that Uzbek people use on daily basis nowadays) in the worst shape and can't be shaped into proper grammar & rules as this language has 12 distributed variations whereas you can't tell which of them is right. The only way to fix this is to reinvent the wheel and force it and this is what we are doing via localization projects & governmental SMs. Btw, the diagram you showed me above is correct & incorrect at the same time. I hope you get what I mean :D

Reply:
(send me I mean post the image here, not personally or anything)

Reply:
> Get one thing right. There are no resources about Uzbek language grammar. This is not systematized. Also, the expression of a sentence in the Uzbek language differs in different dialects. The number of dialects is about one hundred in total. Each village even has its own dialect.
> 
> Mushuk, pishak, mushu, mishig' :)

LMFAO :D I confirm this as a native speaker!

Reply:
Another addition.  There are two different alphabets in the Uzbek language, and separate rules apply to each alphabet.

Reply:
I'm just writing my closing remarks in this review thread. If anybody familiar with translation process can send their dependency or constituency parse of "Tur inferensiyasi" and "Inferensiya turi" (in the assumed alphabet of this translation), it would be helpful for other eyes, but otherwise, I just wish you people good luck with improving digital resources for Uzbek people! Keep at it, open-source helps people ❤️ (and if possible please avoid ad-hominems and curse words to improve constructive quality of the dialogue)

Reply:
1. The one which adopts Russian Cyrillic characters which has additions to meet Uzbek dialects
2. Latin (which current Uzbek language uses) that adopted grammars from Russian based Cyrillic dialect. 

even if latin is being taught at schools, gov still uses cyrillic and follows hella another separate rules which can't be used in public. @mehmetoguzderin now you see the problem?

Reply:
@katsuki-yuri I do recognize the struggle. But as I said in my last comment, I just have a final suggestion for the dialogue, otherwise, everything you all express here are well-noted and understood. Thank you.

Reply:
Hi! I would like to add something regarding the translation of "Type inference". After reading through all the thread, the confusion is in the understanding of the term of "Type Inference" itself in English. You can see the real difference between the terms here in these two examples: "the model is being used to do inference on types [Tur Inferensiyasi]" vs "there are around 10 inference types introduced in the last 2 years [Inferensiya Turlari]" 

@mehmetoguzderin makes a good point in clarifying exactly which meaning of the phrase "Type Inference" being used here and I don't understand how the conversation about Latin and Cyrillic is relevant here at all. 

But anyways, this is a small issue in itself and wouldn't probably make a big difference in the end. Thanks to @katsuki-yuri  and @ismoilovdevml for the initiative!